pkg/storage/etcd/etcd_watcher.go

```diff
@@ -359,9 +359,6 @@ func (w *etcdWatcher) sendAdd(res *etcd.Response) {
 		return
 	}
 	action := watch.Added
-	if res.Node.ModifiedIndex != res.Node.CreatedIndex {
```

Comment:
sendAdd() will be called only within sendResult():
```
func (w *etcdWatcher) sendResult(res *etcd.Response) {
	switch res.Action {
	case EtcdCreate, EtcdGet:
		w.sendAdd(res)
	...
	}
}
```

- EtcdCreate should always have `res.Node.ModifiedIndex != res.Node.CreatedIndex`.
- EtcdGet is from Get(). Get() is only called in etcdGetInitialWatchState() which is only called on watch 0 case in watcher.